### ISSUE TITLE
"norun" cell metadata

### ISSUE THREAD

**Issue by stevengj** [jupyter/nbformat#20]
It occurred to me that it would be useful to have a `norun` code-cell metadata flag, that used to:

* Prevent the cell from being executed in the notebook interface when *Run All* is selected — you have to run the cell explicitly with shift-enter etc.
* Prevent the cell from being executed when it is included in another notebook via the `%run` magic in IPython.
* Prevent the cell from being executed when the notebook is included in other contexts, e.g. in my [NBInclude](https://github.com/stevengj/NBInclude.jl) package for including IJulia notebooks in Julia programs.

I have been thinking about [literate programming](https://en.wikipedia.org/wiki/Literate_programming) styles based on the notebook format, which is why I made NBInclude so that notebooks can be first-class collections of Julia code (as an alternative/supplement to `.jl` files).  If I'm developing code in a notebook, I might want runnable code examples and tests mixed with the "production" code, but I wouldn't want those snippets to be executed along with the rest of the file when I include the file elsewhere.

The UI in the notebook could be a checkable "Skip Cell in Run All" menu item in the `Cell` menu, and perhaps the `norun` cells could be displayed slightly differently, e.g. with a darker-gray background.

**Comment by minrk** [comment]
I think this is a reasonable candidate for a tag. We've been talking about using a list of tags for some simple metadata markers, and this seems like a good one. cc @ellisonbg

**Comment by stevengj** [comment]
Yes, tags are potentially very useful.  e.g. I'd also like to use a `test` tag for marking unit tests, and a `doc` tag to label markdown cells as docstrings for Julia functions and methods.

**Comment by stevengj** [comment]
(There's also the question of whether the cell metadata tags should be sent to the kernel as part of the `execute_request` message.   I don't have any immediate use for this, but I can imagine it could be useful.)

**Comment by rgbkrk** [comment]
> I think this is a reasonable candidate for a tag. We've been talking about using a list of tags for some simple metadata markers, and this seems like a good one. cc @ellisonbg

I've closed my PR that attempted to standardize tags and description: https://github.com/jupyter/nbformat/pull/134

If anyone wants to have at it, feel free.
